packages/agreement-process/src/services/agreementService.ts

```diff
@@ -424,37 +429,33 @@ export function agreementServiceBuilder(
         })
       ).filter((a: WithMetadata<Agreement>) => a.data.id !== agreement.data.id);
 
+      const hasRelatedAgreements = agreements.length > 0;
       const updatedAgreement = {
         ...agreement.data,
         ...updateSeed,
       };
 
-      const contract = await contractBuilder(
+      const contractBuilderInstance = contractBuilder(
         readModelService,
         pdfGenerator,
         fileManager,
         config,
         logger
-      ).createContract(
-        authData.selfcareId,
-        agreement.data,
+      );
+
+      const submittedAgreement = await addContractOnFirstActivation(
+        contractBuilderInstance,
         eservice,
         consumer,
         producer,
-        updateSeed
+        updateSeed,
+        authData,
+        updatedAgreement,
+        hasRelatedAgreements
       );
 
-      const submittedAgreement =
-        updatedAgreement.state === agreementState.active &&
-        agreements.length === 0
-          ? {
-              ...updatedAgreement,
-              contract: apiAgreementDocumentToAgreementDocument(contract),
-            }
-          : updatedAgreement;
-
       const agreementEvent =
-        newState === agreementState.active
+        submittedAgreement.state === agreementState.active
```

Comment:
@galales in this case I have replaced `newState` with state of`submittedAgreement`, it should be the same, in fact `createSubmissionUpdateAgreementSeed` using `newState` to set the state on `submittedAgreement`